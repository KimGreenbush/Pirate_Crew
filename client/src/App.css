.App {
  text-align: center;
  font-family: "Calligraffitti", cursive;
  color:rgb(126, 40, 40);
}

main h1 {
  display: inline-block;
  margin: 20px;
  font-size: 80px;
}

button {
  background-color: rgb(114, 41, 41);
  color: white;
  font-family: "Calligraffitti", cursive;
  border: 1px solid rgb(114, 41, 41);
  border-radius: 50%;
  box-shadow: 1px 1px 3px black;
  width: 120px;
  height: 50px;
  margin: 5px;
}

.pirates {
  margin: 20px;
}

.pirate {
  display: inline-block;
  margin: 30px;
}

.pirate img {
  width: 100px;
}

.pirate-form label {
  display: block;
  margin-top: 10px;
}

.pirate-form button {
  display: block;
  margin: 10px auto;
}

.pirate-about img {
  max-width: 500px;
}

.pirate-about>div {
  display: inline-block;
  vertical-align: top;
  padding: 10px 50px;
}

.pirate-about>.column button {
  width: 50px;
}

@media
only screen and (max-width: 760px){
  .pirate-about img {
    max-width: 100%;
  }
}