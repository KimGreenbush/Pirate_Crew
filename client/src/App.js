import React, { useState } from "react";
import { Router, navigate } from "@reach/router";
import axios from "axios";
import Main from "./views/Main";
import AddForm from "./views/AddForm";
import Show from "./views/Show";

// css import fail tests... why?
// import "./App.css";

function App() {
    const [pirates, setPirates] = useState([]);
	const [errors, setErrors] = useState([]);

	const createPirate = (pirate) => {
		axios.post("http://localhost:9001/api/pirates/new", pirate)
			.then((res) => {
				setPirates([...pirates, res.data])
				setErrors([])
				navigate("/pirates")
			})
			.catch((err) => {
				const errorRes = err.response.data.errors;
				// console.log(errorRes)
				setErrors(Object.keys(errorRes).map((error) => errorRes[error].message))
			})
	}

	return (
		<div className="App">
			<Router>
				<Main path="/pirates" pirates={pirates} setPirates={setPirates} />
				<AddForm path="/pirate/new" onSubmitProp={createPirate} errors={errors} setErrors={setErrors} />
				<Show path="/pirate/:id" />
			</Router>
		</div>
	);
}

export default App;
