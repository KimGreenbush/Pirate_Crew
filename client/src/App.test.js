/**
 * @jest-environment jsdom
 */

import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import renderer from "react-test-renderer";

it("renders App without crashing", () => {
	const div = document.createElement("div");
	ReactDOM.render(<App />, div);
});

test("App get pirates", () => {
	const component = renderer.create(
		<App />,
	)
	let tree = component.toJSON()
	expect(tree).toMatchSnapshot();
})