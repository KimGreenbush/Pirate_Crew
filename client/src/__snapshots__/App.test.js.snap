// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App without crashing 1`] = `
<div
  className="App"
>
  <div
    style={
      Object {
        "outline": "none",
      }
    }
    tabIndex="-1"
  />
</div>
`;
